File: Chapter01/Python_Code/CombineRollingMonths.py
import os
import pandas as pd
from dateutil.relativedelta import relativedelta
    
os.chdir("./Data/SalesData")
monthly_reports = os.listdir(".")

d = {'monthly_reports': monthly_reports}
df = pd.DataFrame(d)
df["date_format"] = pd.to_datetime(df.monthly_reports.str.replace(".csv",""))

min_month = df.date_format.max() - relativedelta(months=12)

reports_to_read = df[df["date_format"]>=min_month]

df_output = pd.concat(map(pd.read_csv, reports_to_read["monthly_reports"]))

File: Chapter03/R_Code/CombineRollingMonths.R
library(tidyverse)
library(lubridate)
library(stringr)

setwd("./Data/SalesData")
monthly_reports <- list.files(".")
date_format <- stringr::str_replace(monthly_reports, ".csv","")
date_format <- lubridate::ymd(date_format)
df <- data_frame(monthly_reports, date_format)

max_month <- max(df$date_format)
min_month <- max_month %m+% months(-12)

reports_to_read <- 
    df$monthly_reports[df$date_format >= min_month]

df_output <- purrr::map_df(reports_to_read, read_csv)
